### PR TITLE
Milestone 2: Feature/interactive list

### PR DESCRIPTION
**Pull request by alvp01**
#In this milestone I:

- Add 2 new JavaScript files and import them as modules: 
  -  one will contain methods related to drag / drop and sorting functionality.
  - the other will contain methods related to the status updates (completed: true / false).
- Add event listeners to list items (dragstart, dragover, drop) and to checkbox (change).
- Update items object's values for index and completed keys upon user actions.
- Store the updated array of items in local storage, so the user gets the correct list values after the page reloads.

### CODE REVIEW

File: src/modules/storage.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const getTaskData = () => JSON.parse(localStorage.getItem('TaskData') || '[]');
+
+export const storeListData = (tasksArray) => {
+  localStorage.setItem('TaskData', JSON.stringify(tasksArray));
```
Comment (MahmoudBakr23) [inline src/modules/storage.js:4]: - [ ] Kindly double check on the local storage because it for somehow resets at the 2nd reload!

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @alvp01 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Linter is correct 
- Descriptive pr summary and headline 
- Great code separation

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @alvp01 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by alvp01** [comment]
> Hi @alvp01 smiley,
> 
> Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
> Congratulations! tada
> 
> Cheers and Happy coding!clapclapclap
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Thank you for your time and work!

**Pull request merged**